src/wp-includes/option.php

```diff
@@ -1093,16 +1106,15 @@ function delete_option( $option ) {
 	$result = $wpdb->delete( $wpdb->options, array( 'option_name' => $option ) );
 
 	if ( ! wp_installing() ) {
-		if ( 'yes' === $row->autoload ) {
+		if ( 'yes' === $row->autoload || 'on' === $row->autoload || 'auto-on' === $row->autoload || 'auto' === $row->autoload ) {
```

Comment:
Nit-pick: We could move `yes` to the end, since it's only supported for BC, and to keep the order consistent.
```suggestion
		if ( 'on' === $row->autoload || 'auto-on' === $row->autoload || 'auto' === $row->autoload || 'yes' === $row->autoload ) {
```